src/utils/wrappers.py

```diff
@@ -0,0 +1,343 @@
+        images = images.filter(ee.Filter.gt('hour_difference', 0))
+        clip_scale_func = clip_and_scale(geometry, scale)
+        images = images.sort('hour_difference', True).map(clip_scale_func)
+        image = images.reduce(ee.Reducer.firstNonNull())
+        image = image.select(["%s_first" % b for b in self.bands], self.bands)
+        return image
+
+
+class WindImagery(CollectionClass):
+    """."""
+
+    def __init__(self, link, bands=None, before_range=(-12, 'hour')):
+        """.
+
+        Parameters
+        ----------
+        link : str
+        bands : list[str] optional
+        before_range : tuple, optional
+            Extent of wind bands to take before givne date.
+            The default is (-12, 'hour').
+        """
+        super(WindImagery, self).__init__(link, bands)
+        self.before = before_range
+
+    def get_bands(self, date, geometry):
+        """Get stacked bands from the wind imagery range.
+
+        Parameters
+        ----------
+        date : ee.Date
+        geometry : ee.Geometry
+
+        Returns
+        -------
+        ee.Image
+            Wind bands where i_b is value of band "b" "i" hours before date.
+
+        """
+        images = self.get_clipped_bands(geometry)
+        images = images.filter(ee.Filter.date(date.advance(*self.before), date)
+                               )
+        images = images.map(lambda img: self._stack_bands(img, date).clip(
+            geometry))
+
+        return merge_bands(images)
+
+    def _stack_bands(self, img, date):
+        """Stack bands from different images, helper method.
+
+        Parameters
+        ----------
+        img : ee.Image
+        date : ee.Date
+
+        Returns
+        -------
+        ee.Image
+            Image with renamed band preficed by time difference in hours
+
+        """
+        return img.select(self.bands,
+                          [ee.Algorithms.String(date.difference(img.date(),
+                                                                'hour').int()
+                                                ).cat('_%s' % b)
+                           for b in self.bands])
+
+
+def merge_bands(images):
+    """Create image from collection by stacking the bands from the images.
+
+    Parameters
+    ----------
+    images : ee.ImageCollection
+        The image collection from which bands will be stacked
+
+    Returns
+    -------
+    ee.Image
+        Image with stacked bands
+
+    """
+    def stack_bands(new, previous):
+        return ee.Image(previous).addBands(new)
+
+    return ee.Image(images.iterate(stack_bands, ee.Image(None)))
+
+
+def clip_and_scale(geometry, scale):
+    """Return clipToBoundsAndScale map function.
+
+    Parameters
+    ----------
+    geometry : ee.Geometry
+    scale : int
+
+
+    Returns
+    -------
+    TYPE
+        DESCRIPTION.
+
+    """
+    def clip_scale(image):
+        return image.clipToBoundsAndScale(geometry, scale=scale)
```

Comment:
In general, you probably don't need to put this small small logic into its own function. 
It would be fine to just call "image.clipToBoundsAndScale(geometry, scale=scale)" wherever it is needed to be called instead of clip_and_scale.

But it is fine to keep this one as is.